### ISSUE TITLE
question about the paper

### ISSUE THREAD

**Issue by Christian-lyc** [agaldran/balanced_mixup#5]
Hi, thank you for your code, it inspires me a lot.
Just one thing confuses me, in the paper, it says 'the learning rate is cyclically annealed towards 0.' But in the code you use ReduceLROnPlateau scheduler. Is this the same one you mentioned in the paper?
However I used CyclicLR scheduler and found it could improve the performance. I'm not sure this is the one similar to paper?

Another thing is in the paper, it says, 'After training, we generate predictions by averaging the result of the input image and its horizontally flipped version, which is common practice to regularize predictions'
In the code, during validate phase, you seem only use original image as input and don't other transformation except resize and totensor 
because I'm using it to other dataset and on more complex model, so it's important to understand the implementation detail.
Thank you again.
Best regards,

**Comment by agaldran** [comment]
Hello Christian,

You are right, in this project apparently I used a patience-based lr decay. Whenever in doubt about paper or code, the code always is the correct version of things.

On the other hand, I checked and the predictions are with Test Time Augmentation (TTA), using horizontal and vertical flips. [Here](https://github.com/agaldran/balanced_mixup/blob/c9e053b95c2400a6c33e3cc1c0b405c37595147c/utils/get_loaders.py#L276) is the definition of the test dataset, which intially has a probability of flipping of p=0. Then during testing [here](https://github.com/agaldran/balanced_mixup/blob/c9e053b95c2400a6c33e3cc1c0b405c37595147c/test_with_labels_endo.py#L78C13-L78C41) that probability is switched on and off, and the angles of rotation are varied, so this thing ends up flipping the image in several directions. A bit convoluted, but the end result is the flipping of the input tensors.

Cheers,

Adrian
